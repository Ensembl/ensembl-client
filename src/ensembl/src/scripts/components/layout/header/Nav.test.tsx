import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Nav } from './Nav';

describe('<Nav />', () => {
  let launchbarFn: () => void;
  let accountFn: () => void;
  let wrapper: any;

  beforeEach(() => {
    launchbarFn = jest.fn();
    accountFn = jest.fn();
    wrapper = shallow(
      <Nav toggleAccount={accountFn} toggleLaunchbar={launchbarFn} />
    );
  });

  test('has two buttons', () => {
    expect(wrapper.find('button')).toHaveLength(2);
  });

  describe('should toggle', () => {
    test('launchbar on launchbar button click', () => {
      wrapper.find('.launchbar-button').simulate('click');

      expect(launchbarFn).toHaveBeenCalled();
    });

    test('account on account button click', () => {
      wrapper.find('.account-button').simulate('click');

      expect(accountFn).toHaveBeenCalled();
    });
  });

  test('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
