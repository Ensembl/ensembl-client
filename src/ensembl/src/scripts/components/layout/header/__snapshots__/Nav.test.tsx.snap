// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> renders correctly 1`] = `
<div
  className="top-bar-right"
>
  <button
    className="inline launchbar-button"
    onClick={[MockFunction]}
  >
    <img
      alt="toggle launchbar"
      src={Object {}}
      title="Launchbar"
    />
  </button>
  <button
    className="inline account-button"
    onClick={[MockFunction]}
  >
    <img
      alt="toggle account"
      src={Object {}}
      title="Account"
    />
  </button>
</div>
`;
